cni/pkg/iptables/iptables.go

```diff
@@ -72,11 +74,37 @@ func NewIptablesConfigurator(cfg *Config, ext dep.Dependencies, nlDeps NetlinkDe
 			RestoreFormat: true,
 		}
 	}
-	return &IptablesConfigurator{
+
+	configurator := &IptablesConfigurator{
 		ext:    ext,
 		nlDeps: nlDeps,
 		cfg:    cfg,
 	}
+
+	// By detecting iptables versions *here* once-for-all we are
+	// committing to using the same binary/variant (legacy or nft)
+	// within all pods as we do on the host.
+	//
+	// This should be fine, as the host binaries are all we have to work with here anyway,
+	// as we are running within a privileged container - and we don't want to take the time to
+	// redetect for each pod anyway.
+	//
+	// Extreme corner case:
+	// If for some reason your host had both binaries, and you were injecting out-of-band
+	// iptables rules within a pod context into `legacy` tables, but your host context preferred
+	// `nft`, we would still inject our rules in-pod into nft tables, which is a bit wonky.
+	//
+	// But that's stunningly unlikely (and would still work either way)
+	iptVer, err := ext.DetectIptablesVersion(false)
```

Comment:
We should probably log the error messages as debug logs.

Reply:
Yes, I agree and I don't like this. I prefer not to call error-able funcs from init funcs. I did here to avoid addl refactoring.

I think I will just change the init func to return an error - it's more explicit and the caller should handle logging anyway.